### PR TITLE
Crash Course tutorial for datasets and dataloader 

### PR DESCRIPTION
**Pull request by jamesgolden1**
Mostly the same as previous crash course with C++ backend dataloader demo added at the end
Thanks to MXNet team member zhreshold for building and documenting this.

## Description ##
As part of the MXNet 2.0 crash course, this is the section on datasets and dataloaders. There are not many big changes in terms of dataloaders, so I used the previous dataloaders crash course tutorial and added a section at the end on the C++ backend dataloader.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: docs/python_docs/python/tutorials/getting-started/crash-course/5-datasets.md [hunk 1]
```diff
@@ -0,0 +1,543 @@
...
+    Requirement already satisfied: certifi>=2017.4.17 in /home/ec2-user/anaconda3/envs/mxnet_p36/lib/python3.6/site-packages (from requests<3,>=2.20.0->mxnet) (2020.6.20)
+    Building wheels for collected packages: contextvars
+      Building wheel for contextvars (setup.py) ... [?25ldone
+    [?25h  Created wheel for contextvars: filename=contextvars-2.4-py3-none-any.whl size=7664 sha256=c4195e935f5505bed92acbaa71e886731fa08e6496d9dc8de3ee4453cf5613ec
+      Stored in directory: /home/ec2-user/.cache/pip/wheels/41/11/53/911724983aa48deb94792432e14e518447212dd6c5477d49d3
+    Successfully built contextvars
+    Installing collected packages: immutables, contextvars, mxnet
+    Successfully installed contextvars-2.4 immutables-0.14 mxnet-2.0.0b20200831
+    [33mWARNING: You are using pip version 20.0.2; however, version 20.2.2 is available.
+    You should consider upgrading via the '/home/ec2-user/anaconda3/envs/mxnet_p36/bin/python -m pip install --upgrade pip' command.[0m
```
Comment (leezu) [inline docs/python_docs/python/tutorials/getting-started/crash-course/5-datasets.md]: In the 5th chapter of the crashcoure I think we can assume that mxnet is installed

File: docs/python_docs/python/tutorials/getting-started/crash-course/5-datasets.md [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+
+    def __next__(self):
+        batch = self.data_iter.__next__()
+        assert len(batch.data) == len(batch.label) == 1
+        data = batch.data[0]
+        label = batch.label[0]
+        return data, label
+
+    def next(self):
+        return self.__next__() # for Python 2
```
Comment (leezu) [inline docs/python_docs/python/tutorials/getting-started/crash-course/5-datasets.md]: Python 2 is no longer supported

File: docs/python_docs/python/tutorials/getting-started/crash-course/5-datasets.md [hunk 3]
```diff
@@ -0,0 +1,543 @@
...
+```
+
+Performance concerns include slow python dataset/transform functions, multithreading issues due to global interpreter lock, Python multiprocessing issues due to speed, and batchify issues due to poor memory management.
+
+This new dataloader provides: 
+- common C++ batchify functions that are split and context aware
+- a C++ MultithreadingDataLoader which inherit the same arguments as gluon.data.DataLoader but use MXNet internal multithreading rather than python multiprocessing.
+- fallback to python multiprocessing whenever the dataset is not fully supported by backend(e.g., there are custom python datasets)
+- the transform is not fully hybridizable
+- batchify is not fully supported by backend
```
Comment (leezu) [inline docs/python_docs/python/tutorials/getting-started/crash-course/5-datasets.md]: The last two items seem to belong to the fallback item?

**Comment by jamesgolden1** [comment]
Wrong branch - closing this because I made a different PR on the correct branch.
